flocker/node/test/test_script.py

```diff
@@ -462,44 +467,213 @@ def test_starts_convergence_loop(self):
                           P2PNodeDeployer, b"1.2.3.4", service, True))
 
 
-class ZFSAgentOptionsTests(make_volume_options_tests(
-        ZFSAgentOptions, [b"1.2.3.4", b"example.com"])):
+class DatasetAgentServiceFactoryTests(SynchronousTestCase):
     """
-    Tests for the volume configuration arguments of ``ZFSAgentOptions``.
+    Tests for ``DatasetAgentServiceFactory``.
     """
-    def test_default_port(self):
+    def test_get_service(self):
         """
-        The default AMP destination port configured by ``ZFSAgentOptions`` is
-        4524.
+        ``DatasetAgentServiceFactory.get_service`` creates ``AgentLoopService``
+        configured with the destination given by the options.
         """
-        options = ZFSAgentOptions()
-        options.parseOptions([b"1.2.3.4", b"example.com"])
```

Comment:
I half-feel most of these should be functional tests, not unit tests. But they're already written, so that's fine.